analysis.go

```diff
@@ -155,9 +156,18 @@ func ListPackages(fileRoot, importRoot string) (PackageTree, error) {
 		ip := filepath.ToSlash(filepath.Join(importRoot, strings.TrimPrefix(path, fileRoot)))
 
 		// Find all the imports, across all os/arch combos
-		p, err := ctx.ImportDir(path, analysisImportMode())
+		ap, err := filepath.Abs(path)
+		if err != nil {
+			return err
+		}
+		p, err := fullPackageInDir(ap)
+		//fmt.Printf("fullPackage(%q) == err=%q == %#v\n", ap, err, p)
+		//p, err := ctx.ImportDir(path, analysisImportMode())
 		var pkg Package
 		if err == nil {
+			if p.Name == "" {
+				p.Name = importRoot
```

Comment:
seems weird that`p.Name` could be empty? if so, though, then it should probably be assigned `path.Base(importRoot)`, so that it's just getting the last element.